apps/client/src/app/pages/voyage-tracker/voyage-tracker/vessel-row/vessel-rank-column/vessel-rank-column.component.ts

```diff
@@ -0,0 +1,32 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { FreeCompanyWorkshopFacade } from '../../../../../modules/free-company-workshops/+state/free-company-workshop.facade';
+import { observeInput } from '../../../../../core/rxjs/observe-input';
+
+@Component({
+  selector: 'app-vessel-rank-column',
+  templateUrl: './vessel-rank-column.component.html',
+  styleUrls: ['./vessel-rank-column.component.less'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class VesselRankColumnComponent {
+  @Input() rank: number;
+
+  vessel$ = observeInput(this, 'rank');
+
+  @Input() currentExperience: number;
+
+  currentExperience$ = observeInput(this, 'currentExperience');
+
+  @Input() totalExperienceForNextRank: number;
+
+  totalExperienceForNextRank$ = observeInput(this, 'totalExperienceForNextRank');
+
+  @Input() maxRank: number;
+
+  constructor(private freeCompanyWorkshopFacade: FreeCompanyWorkshopFacade) {
+  }
+
+  rankUpPercent(): number {
+    return Math.round(100 * (this.currentExperience / this.totalExperienceForNextRank));
+  }
```

Comment:
You should make use of these nice `Observable`s above instead of making a method to then call it in template (as explained above, bad practice etc).

I'm going to write the `Observable` here with comments so you can understand it nicely:

```ts
// combineLatest because we want to only compute again if one of the observables emitted a new value, combined 
// with the latest value from the other one
rankupPercent$ = combineLatest([this.currentExperience$, this.totalExperienceForNextRank$]).pipe(
  // map will allow us to project these two values into tyhe computed value that we want.
   map(([currentExperience, totalExperienceForNextRank]) => {
    // Your original formula, but this time it'll only be executed if one of the inputs changes, not every frame that the change detector will compute.
    return Math.round(100 * (currentExperience / totalExperienceForNextRank));
   })
)
```

Then you can use it by doing `rankupPercent$ | async` instead of `rankupPercent()` in the template.

Remember to delete the `Observable` fields that you're not using to avoid creating `Observable`s for nothing 🙂.